File: tests/src/com.esotericsoftware/kryo-shaded/3.0.3/src/test/java/com_esotericsoftware/kryo_shaded/JavaSerializerTest.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package com_esotericsoftware.kryo_shaded;

import static org.assertj.core.api.Assertions.assertThat;

import com.esotericsoftware.kryo.Kryo;
import com.esotericsoftware.kryo.io.Input;
import com.esotericsoftware.kryo.io.Output;
import com.esotericsoftware.kryo.serializers.JavaSerializer;
import java.io.Serializable;
import org.junit.jupiter.api.Test;

public class JavaSerializerTest {
    @Test
    void serializesSerializableObjectThroughJavaObjectStreams() {
        Kryo writer = createKryo();
        JavaSerializableValue original = new JavaSerializableValue("alpha", 42);

        Output output = new Output(128, -1);
        writer.writeObject(output, original);
        output.close();

        Kryo reader = createKryo();
        JavaSerializableValue restored = reader.readObject(new Input(output.toBytes()), JavaSerializableValue.class);

        assertThat(restored).isNotSameAs(original);
        assertThat(restored.getName()).isEqualTo(original.getName());
        assertThat(restored.getCount()).isEqualTo(original.getCount());
    }

    private static Kryo createKryo() {
        Kryo kryo = new Kryo();
        kryo.register(JavaSerializableValue.class, new JavaSerializer());
        return kryo;
    }

    public static class JavaSerializableValue implements Serializable {
        private static final long serialVersionUID = 1L;

        private final String name;
        private final int count;

        JavaSerializableValue(String name, int count) {
            this.name = name;
            this.count = count;
        }

        public String getName() {
            return name;
        }

        public int getCount() {
            return count;
        }
    }
}

File: tests/src/org.apache.pekko/pekko-actor_2.13/1.6.0/src/test/scala/org_apache_pekko/pekko_actor_2_13/JavaSerializerTest.scala
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package org_apache_pekko.pekko_actor_2_13

import java.io.Serializable

import scala.concurrent.Await
import scala.concurrent.duration.DurationInt

import com.typesafe.config.Config
import com.typesafe.config.ConfigFactory
import org.apache.pekko.actor.ActorSystem
import org.apache.pekko.actor.ExtendedActorSystem
import org.apache.pekko.serialization.JavaSerializer
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class JavaSerializerTest {
  @Test
  def roundTripsSerializableObjectThroughJavaSerializer(): Unit = {
    val system: ActorSystem = ActorSystem("java-serializer-test", javaSerializationConfig())
    try {
      val serializer: JavaSerializer = new JavaSerializer(system.asInstanceOf[ExtendedActorSystem])
      val original: JavaSerializerTest.SerializableMessage =
        JavaSerializerTest.SerializableMessage("pekko-java-serializer", 42)

      val bytes: Array[Byte] = serializer.toBinary(original)
      val restored: AnyRef = serializer.fromBinary(bytes, None)

      assertThat(restored).isEqualTo(original)
    } finally {
      Await.result(system.terminate(), 10.seconds)
    }
  }

  private def javaSerializationConfig(): Config = {
    ConfigFactory.parseString("""
      pekko.actor.allow-java-serialization = on
      pekko.actor.warn-about-java-serializer-usage = off
      pekko.loggers = ["org.apache.pekko.event.Logging$DefaultLogger"]
      """)
  }
}

object JavaSerializerTest {
  @SerialVersionUID(1L)
  final case class SerializableMessage(name: String, value: Int) extends Serializable
}
